src/redis_info_provider/local_watcher.py

```diff
@@ -36,8 +37,17 @@ def __init__(self, update_frequency=1.0):
         """
         self._update_freq = update_frequency
         self._should_stop = gevent.event.Event()
+        self._wait_on = None
         self._greenlet = gevent.spawn(self._greenlet_main)
 
+    def set_wait(self, wait_on):
```

Comment:
I think this event actually only applies when an unhandled exception occurs, right? So maybe something more like `set_exception_event` would be a better name? (And rename the argument accordingly.)